bin/sam-coverage-depth.py

```diff
@@ -281,24 +369,31 @@ def sortKey(base):
     }
 
 
-def main():
+def main() -> None:
     args = makeParser().parse_args()
 
     if not (args.printOffsets or args.printStats):
-        exit("You used both --noOffsets and --noStats, so there is no output!")
+        sys.exit("You used both --noOffsets and --noStats, so there is no output!")
 
     referenceLens = samReferenceLengths(args.samfile)
     referenceId = getReferenceId(args.samfile, set(referenceLens), args.referenceId)
     referenceLen = referenceLens[referenceId]
     referenceSeq = getReferenceSeq(args.fasta, referenceId)
-    baseCounts = []
-    writerow = csv.writer(sys.stdout, delimiter=args.sep).writerow if args.tsv else None
+    depths = []
+    printTSV = csv.writer(sys.stdout, delimiter=args.sep).writerow if args.tsv else None
+    printRow = printTSV or print
 
-    if writerow:
-        row = ["Site"]
+    if printTSV:
+        header = ["Site"]
         if referenceSeq:
-            row.extend(("Reference", "Mutations"))
-        writerow(row + list(BASES))
+            header.append("Reference")
+        header.append("Depth")
+        header.extend(BASES)
+        header.append("Mutations")
+        header.append("Variability")
+        header.extend(f"{pair[0]}->{pair[1]}" for pair in TRANSITIONS + TRANSVERSIONS)
+        header.extend(("Transitions", "Transversions"))
```

Comment:
I heard it's better to use a single call to extend.
Though in this case I guess you'd have to use `chain`, but since that still avoids the attribute lookup it is probably still more performant.

Reply:
Yeah, I thought about this (and the list grew over time). I've converted it to one call to `extend`, which involves making four new lists :-)